src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -547,6 +559,17 @@ public final void doHandle(HttpServletRequest baseRequest,
             }
         }
 
+        // Validate container name
+        if (!uri.equals("/") && !isValidContainer(path[1])) {
+            if (method.equals("PUT") &&
+                    (path.length <= 2 || path[2].isEmpty()) &&
+                    !("".equals(request.getParameter("acl"))))  {
```

Comment:
It can return null( if "acl" is not present), But, we are here checking for an empty "acl" value. Please [refer](https://github.com/andrewgaul/s3proxy/blob/742820a93d4d87e3ff249ae85e92f3eaf405db6e/src/main/java/org/gaul/s3proxy/S3ProxyHandler.java#L622)